src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,39 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final int DATE_INDEX = 0;
+    private static final int NAME_INDEX = 1;
+    private static final int WORKING_HOURS_INDEX = 2;
+    private static final int SALARY_PER_HOUR_INDEX = 3;
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        int totalSalary;
+        LocalDate workingDateFrom = LocalDate.from(FORMATTER.parse(dateFrom));
+        LocalDate workingDateTo = LocalDate.from(FORMATTER.parse(dateTo));
+        StringBuilder employeeSalary = new StringBuilder("Report for period " + dateFrom
+                + " - " + dateTo);
+        for (String name : names) {
+            totalSalary = 0;
+            for (String dataField : data) {
+                String[] dataFields = dataField.split(" ");
```

Comment:
delimiter/separator better be constant too

Reply:
Just FYI mentors probably won't see comments in PR conversations :) No, if  a task is approved nothing else is required